kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,672 @@
+     */
+    void UpdateExternalLoads()
+    {
+        // Update External Loads
+        for (unsigned int i = 0; i < mVariableNames.size(); i++) {
+            ModelPart& r_sub_model_part = *(mSubModelPartList[i]);
+            const std::string& r_variable_name = mVariableNames[i];
+
+            if (KratosComponents<Variable<double>>::Has(r_variable_name)) {
+                const Variable<double>& var = KratosComponents<Variable<double>>::Get(r_variable_name);
+                block_for_each(r_sub_model_part.Nodes(), [&](Node<3>& r_node){
+                    r_node.FastGetSolutionStepValue(var) *= (mLambda/mLambda_old);
+                });
+            }
+            else if (KratosComponents<Variable<array_1d<double,3>>>::Has(r_variable_name)) {
+                typedef Variable<array_1d<double,3>> array_type;
+                const array_type& r_var = KratosComponents<array_type>::Get(r_variable_name);
+
+                block_for_each(r_sub_model_part.Conditions(), [&](Condition& r_condition) {
+                    /* we are multipying by Lambda instead of Lambda/Lambda_old because
+                    the load processes reset the loads to its initial values
+                    when the InitSolStep is called */
+                    if (mInsideIterationLoop)
+                        r_condition.GetValue(r_var) *= mLambda / mLambda_old;
+                    else
+                        r_condition.GetValue(r_var) *= mLambda;
+                });
+
+                // TODO-> add for node loads
+
+            } else {
+                KRATOS_THROW_ERROR( std::logic_error, "One variable of the applied loads has a non supported type. Variable: ", r_variable_name )
+            }
+        }
+
+        // Save the applied Lambda factor
+        mLambda_old = mLambda;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const override
+    {
+        Parameters default_parameters = Parameters(R"(
+        {
+            "name"                                 : "arc_length_strategy",
+            "use_old_stiffness_in_first_iteration" : false,
+            "max_iteration"                        : 10,
+            "reform_dofs_at_each_step"             : false,
+            "compute_reactions"                    : false,
+            "builder_and_solver_settings"          : {},
+            "convergence_criteria_settings"        : {},
+            "linear_solver_settings"               : {},
+            "scheme_settings"                      : {},
+            "desired_iterations"                   : 4,
+            "max_radius_factor"                    : 10.0,
+            "min_radius_factor"                    : 0.1,
+            "loads_sub_model_part_list"            : [],
+            "loads_variable_list"                  : []
```

Comment:
please remove duplicated settings aka settings that already exist in the base-classes

=> Overall the idea how this should work is the same as we use in Python with the `GetDefaultParameters` => you only need to add the new settings in your derived class